### PR TITLE
[FLINK-35234][hotfix][cdc-common] Fix NullPointerException of org.apache.flink.cdc.common.configuration.ConfigurationUtils#convertToString

### PR DESCRIPTION
**Pull request by loserwang1024**
Fix exception like this:
```java
Caused by: java.lang.NullPointerException    at org.apache.flink.cdc.common.configuration.ConfigurationUtils.convertToString(ConfigurationUtils.java:133) ~[?:?]    at org.apache.flink.cdc.common.configuration.Configuration.toMap(Configuration.java:138) ~[?:?] 

```

### CODE REVIEW

**Comment by loserwang1024** [comment]
@yuxiqian , @PatrickRen , a mirror fix, CC

File: flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java [hunk 1]
```diff
@@ -130,7 +130,9 @@ static Duration convertToDuration(Object o) {
     }
 
     static String convertToString(Object o) {
-        if (o.getClass() == String.class) {
+        if (o == null) {
```
Comment (yuxiqian) [inline flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java:133]: Maybe we can bring the check to `convertValue` function? Seems other `convertToXXX` functions also share this problem.

Comment (loserwang1024) [reply flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java:144]: you are right, just done it.

**Review by yuxiqian — APPROVED** [review]
Thanks for @loserwang1024's patch, LGTM.

File: flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java (hunk 1 above)
Comment (Jiabao-Sun) [reply flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java:144]: The inner null check seems redundant now.

Comment (Jiabao-Sun) [reply flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java:144]: > The inner null check seems redundant now.

@loserwang1024 WDYT?

**Review by leonardBang — COMMENTED** [review]
@loserwang1024 Thanks for the contribution, a unit test is welcome

File: flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java (hunk 1 above)
Comment (loserwang1024) [reply flink-cdc-common/src/main/java/org/apache/flink/cdc/common/configuration/ConfigurationUtils.java:144]: > > The inner null check seems redundant now.

Maybe a java.lang.IllegalArgumentException with description is better than NullPointerException?

**Review by leonardBang — APPROVED** [review]
+1

**Pull request merged**